### PR TITLE
decode DispatchInfo on ExtrinsicFailed events

### LINKED ISSUE #186: Concurrent calls return `Scale codec error: No such variant in enum Phase`
When two clients concurrently call the same RPC function, they both return `XtError(Codec(Error("No such variant in enum Phase")))`. I'm able to reliably reproduce this with the code below. The `do_accept_replace` function just calls an RPC function and prints the return value. When I uncomment the delay, both of these functions behave correctly.

```
let task1 = do_accept_replace(client1.clone());
let task2 = do_accept_replace(client2.clone());
tokio::spawn(async move {
    task1.await;
});
// tokio::time::delay_for(tokio::time::Duration::from_secs(10)).await;
tokio::spawn(async move {
    task2.await;
});
```

Versions from Cargo.toml:
```
sp-core = "2.0.0"
frame-system = "2.0.0"
substrate-subxt-proc-macro = { git = "https://github.com/paritytech/substrate-subxt", branch = "master" }
substrate-subxt = { git = "https://github.com/paritytech/substrate-subxt", branch = "master" }
substrate-subxt-client = "0.4.0"
```

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Closes #186

### CODE REVIEW

File: src/events.rs [hunk 1]
```diff
@@ -245,16 +251,22 @@ impl<T: System> EventsDecoder<T> {
...
                 Err(err) => return Err(err),
             };
 
-            r.push((phase, raw));
+            if event_errors.len() == 0 {
+                r.push((phase.clone(), raw));
+            }
+
+            for err in event_errors {
+                r.push((phase.clone(), Raw::Error(err)));
```
Comment (ascjones) [inline src/events.rs:262]:
```suggestion
                r.push((phase, Raw::Error(err)));
```

I guess the second clone is redundant

File: src/events.rs [hunk 2]
```diff
@@ -281,9 +306,83 @@ mod tests {
...
                 &mut output,
+                &mut errors,
             )
             .unwrap();
 
         assert_eq!(output, vec![1, 0]);
     }
+
+    #[test]
+    fn test_decode_system_events_and_error() {
```
Comment (ascjones) [inline src/events.rs:317]: Awesome thanks for the test!

File: src/events.rs [hunk 3]
```diff
@@ -281,9 +306,83 @@ mod tests {
...
+                    extrinsic: ExtrinsicMetadata {
+                        version: 0,
+                        signed_extensions: vec![],
+                    },
+                }),
+            ))
+            .unwrap(),
+        );
+
+        let input = hex::decode("0c00000000000000482d7c0900000000020000000100000000010300035884723300000000000000000200000000010300035884723300000000000000").unwrap();
```
Comment (ascjones) [inline src/events.rs:385]: Can you add a comment describing what this encoded value looks like in a human readable form

**Review by ascjones — COMMENTED** [review]
Looks great! 

Before merging this I need to check what the status of our CLA bot is.

Also could you please include a short description of the bug as you describe in #186

File: src/events.rs (hunk 1 above)
Comment (gregdhill) [reply src/events.rs:262]: The value is moved in the previous iteration of this loop.

**Comment by gregdhill** [comment]
@ascjones I added a comment [here](https://github.com/paritytech/substrate-subxt/pull/189/files#diff-59e0fecb904d610f27811d416aa3f26cc0d1bfed220ac69f8e18147f9bd68fd0R200). Let me know if I should note the issue elsewhere.

**Review by ascjones — APPROVED** [review]
LGTM, just waiting to hear about the CLA from @kirushik before I merge.

**Comment by gregdhill** [comment]
@ascjones can this be merged yet?

**Comment by ascjones** [comment]
Just got the go ahead from @kirushik.

**Pull request merged**